packages/mask/src/plugins/RealityCards/SNSAdaptor/MarketView.tsx

```diff
@@ -0,0 +1,356 @@
+                {t('loading')}
+                <CircularProgress sx={{ mx: 1 }} color="primary" size={13} />
+            </Typography>
+        )
+
+    if (error)
+        return (
+            <Box display="flex" flexDirection="column" alignItems="center">
+                <Typography color="textPrimary">{t('plugin_realitycards_error_something_went_wrong')}</Typography>
+                <Button sx={{ marginTop: 1 }} size="small" onClick={retry}>
+                    {t('retry')}
+                </Button>
+            </Box>
+        )
+
+    if (!market) return null
+    return (
+        <Card variant="outlined" className={classes.root} elevation={0}>
+            <CardHeader
+                title={
+                    <Link target="_blank" href={props.link}>
+                        {market.name}
+                    </Link>
+                }
+                titleTypographyProps={{ variant: 'h5', color: 'textPrimary' }}
+                href={props.link}
+                subheader={<MarketDetails market={market} />}
+                subheaderTypographyProps={{ variant: 'body2' }}
+            />
+
+            <CardContent className={classes.content}>
+                <Grid container direction="column">
+                    {market.cards.map((card: RCCard) => (
+                        <Grid item key={card.id}>
+                            <CardView card={card} market={market} />
+                        </Grid>
+                    ))}
+                </Grid>
+            </CardContent>
+        </Card>
+    )
+}
+
+function MarketDetails(props: MarketDetailsProps) {
+    const { market } = props
+    const { t } = useI18N()
+
+    const [descriptionDialogOpen, setDescriptionDialogOpen] = useState(false)
+    const [giveawayDialogOpen, setGiveawayDialogOpen] = useState(false)
+    const chainId = useChainId()
+    const token = useBaseToken()
+
+    const [day, setDay] = useState(0)
+    const [hour, setHour] = useState(0)
+    const [minute, setMinute] = useState(0)
+    const state: {
+        label: string
+        color: 'default' | 'error' | 'success' | 'warning' | 'primary' | 'secondary' | 'info' | undefined
+        potSize: string
+    } = useMemo(() => {
+        return market.state === MarketState.Open
+            ? {
+                  label: t('plugin_realitycards_event_state_open'),
+                  color: 'success',
+                  potSize: t('plugin_realitycards_event_pot_size'),
+              }
+            : market.state === MarketState.Withdraw
+            ? {
+                  label: t('plugin_realitycards_event_state_ended'),
+                  color: 'default',
+                  potSize: t('plugin_realitycards_event_final_pot_size'),
+              }
+            : {
+                  label: t('plugin_realitycards_event_state_locked'),
+                  color: 'warning',
+                  potSize: t('plugin_realitycards_event_final_pot_size'),
+              }
+    }, [market.state])
```

Comment:
Don't nest ternary expressions, it's hard to understand at a glance.
```suggestion
    } = useMemo(() => {
        if (market.state === MarketState.Open) {
          return {
                  label: t('plugin_realitycards_event_state_open'),
                  color: 'success',
                  potSize: t('plugin_realitycards_event_pot_size'),
              }
        }
        if (market.state === MarketState.Withdraw) {
          return {
            abel: t('plugin_realitycards_event_state_ended'),
                  color: 'default',
                  potSize: t('plugin_realitycards_event_final_pot_size'),
          }
        }
       return {
                  label: t('plugin_realitycards_event_state_locked'),
                  color: 'warning',
                  potSize: t('plugin_realitycards_event_final_pot_size'),
              }
    }, [market.state, t])
```

Reply:
Tnx! This is much more readable.